### PR TITLE
Refactor project structure and appearance

### PR DESCRIPTION
**Pull request by MaksimkaSota**
1. Refactored project structure and appearance;
2. Added destructuring for the useTypedSelector;
3. Added a project information in the README file;
4. Added the LoginInformation component with a copy text logic.

### CODE REVIEW

File: src/Components/Common/ErrorCatcher/ErrorCatcher.tsx [hunk 1]
```diff
@@ -1,23 +1,29 @@
...
+  state: StateType = {
+    UIError: null,
+    promiseError: null,
+  };
 
   static getDerivedStateFromError() {
-    return { errorMessage: 'Some UI Error! We are sorry... Fix it soon!' };
+    return {
+      UIError: {
+        message: 'Some UI Error! We are sorry... Fix it soon!',
```
Comment (ZhekaVasil) [inline src/Components/Common/ErrorCatcher/ErrorCatcher.tsx]: Be more polite here =)
We are sorry... We will fix it soon! or 
We are sorry... The fix is coming!

File: src/Components/Main/Pages/DialogsPage/DialogsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import type { FC, ReactElement } from 'react';
+import classes from './DialogsPage.module.scss';
+import { Messages } from './Messages/Messages';
+import type { MessageType, PostType } from '../../../../utils/types/common';
+import { Posts } from './Posts/Posts';
+
+type PropsType = {
+  messages: Array<MessageType>;
```
Comment (ZhekaVasil) [inline src/Components/Main/Pages/DialogsPage/DialogsPage.tsx]: Just FYI, you could use short version:
messages: MessageType[];

**Review by ZhekaVasil — APPROVED** [review]

File: src/Components/Common/ErrorCatcher/ErrorCatcher.tsx (hunk 1 above)
Comment (MaksimkaSota) [reply src/Components/Common/ErrorCatcher/ErrorCatcher.tsx]: Done!
I  am just learning English... :)

File: src/Components/Main/Pages/DialogsPage/DialogsPage.tsx (hunk 2 above)
Comment (MaksimkaSota) [reply src/Components/Main/Pages/DialogsPage/DialogsPage.tsx]: This is not necessary, but I have corrected it!

**Pull request merged**